### PR TITLE
clean up database directory on failure of open

### PR DESCRIPTION
**Pull request by muralivemulapati**
rocksdb::DB* db;
rocksdb::Options options;
options.create_if_missing = false;
rocksdb::Status status = rocksdb::DB::Open(options, "/tmp/testdb", &db);

If /tmp/testdb did not exist before DB::Open(), it leaves a database directory
with two files LOG and LOCK after return from DB::Open().

### CODE REVIEW

File: db/db_impl_open.cc [hunk 1]
```diff
@@ -347,6 +347,13 @@ Status DBImpl::Recover(
           return s;
         }
       } else {
+        env_->DeleteFile(LockFileName(dbname_));
+        std::string db_absolute_path;
+        env_->GetAbsolutePath(dbname_, &db_absolute_path);
+        std::string logfname =
+            InfoLogFileName(dbname_, db_absolute_path, immutable_db_options_.db_log_dir);
+        env_->DeleteFile(logfname);
```
Comment (anand1976) [inline db/db_impl_open.cc]: I don't think we should delete the log file unless we're sure this call to ```DB::Open()``` created it.

Comment (muralivemulapati) [reply db/db_impl_open.cc]: Thanks. Taken care of this by keeping track of created directories and cleaning up on failure.

**Comment by siying** [comment]
@anand1976 do you remember where we are?

**Comment by muralivemulapati** [comment]
Thanks for the feedback. I will resubmit PR with the suggested code changes.

File: db/db_impl.h [hunk 2]
```diff
@@ -771,6 +771,14 @@ class DBImpl : public DB {
   std::unique_ptr<VersionSet> versions_;
   // Flag to check whether we allocated and own the info log file
   bool own_info_log_;
+  // Keep track of directories created during Open(). These will be deleted
+  // if Open() fails.
+  std::string created_info_log_file_;
```
[the remark this answers is not in the document]
Comment (muralivemulapati) [reply db/db_impl.h]: Implemented the suggested changes.

**Comment by anand1976** [comment]
@muralivemulapati After the rebase, I'm seeing a lot of commits and 676 changed files, making it very hard to review. You should probably create the PR from a branch in your fork, rather than master, to avoid this problem.

**Comment by muralivemulapati** [comment]
This is the new pull request: 5819.

**Pull request closed without merging**